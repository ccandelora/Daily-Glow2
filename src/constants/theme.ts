export const COLORS = {
  // Primary colors for emotions (keeping these for emotional context)
  primary: {
    red: '#FF6B6B',     // Warm red for negative emotions
    green: '#00C853',   // Vibrant green for positive emotions
    blue: '#2979FF',    // Bright blue for calm emotions
    yellow: '#FFD600',  // Bright yellow for energetic emotions
  },
  
  // UI colors updated for dark theme with glowing accents
  ui: {
    background: 'rgba(28, 14, 45, 0.95)',  // Dark purple background
    card: 'rgba(38, 20, 60, 0.85)',        // Slightly lighter purple for cards
    text: '#FFFFFF',                        // White text for contrast
    textSecondary: 'rgba(255, 255, 255, 0.7)', // Semi-transparent white
    border: 'rgba(82, 67, 194, 0.3)',      // Subtle purple border
    disabled: 'rgba(82, 67, 194, 0.2)',    // Dimmed purple for disabled state
    accent: '#4169E1',                     // Royal blue for accents
    highlight: 'rgba(65, 105, 225, 0.15)', // Soft blue highlight
  },

  // Status colors with glow effects
  status: {
    success: '#00E676',
    error: '#FF5252',
    warning: '#FFD600',
    info: '#4169E1',
  },

  // Gradient colors for glow effects
  gradient: {
    start: 'rgba(65, 105, 225, 0.2)',  // Royal blue glow
    middle: 'rgba(147, 112, 219, 0.15)', // Purple glow
    end: 'rgba(28, 14, 45, 0.9)',     // Dark purple base
  },
};

export const FONTS = {
  sizes: {
    xs: 12,
    sm: 14,
    md: 16,
    lg: 22,
    xl: 28,
    xxl: 36,
    xxxl: 44,
  },
  weights: {
    regular: '400' as const,
    medium: '500' as const,
    semibold: '600' as const,
    bold: '700' as const,
  },
  // Adding font families for more sophistication
  families: {
    heading: 'System',
    body: 'System',
  }
};

export const SPACING = {
  xs: 4,
  sm: 8,
  md: 16,
  lg: 24,
  xl: 32,
  xxl: 40,
};

export const BORDER_RADIUS = {
  sm: 4,
  md: 8,
  lg: 16,
  xl: 24,
  circle: 999,
};

export const TIME_PERIODS = {
  MORNING: {
    label: 'Morning',
    range: { start: 5, end: 11 }, // 5:00 AM - 11:59 AM
    greeting: 'Good morning',
    icon: 'üåÖ'
  },
  AFTERNOON: {
    label: 'Afternoon',
    range: { start: 12, end: 16 }, // 12:00 PM - 4:59 PM
    greeting: 'Good afternoon',
    icon: '‚òÄÔ∏è'
  },
  EVENING: {
    label: 'Evening',
    range: { start: 17, end: 4 }, // 5:00 PM - 4:59 AM
    greeting: 'Good evening',
    icon: 'ÔøΩÔøΩ'
  }
} as const;

export default {
  COLORS,
  FONTS,
  SPACING,
  BORDER_RADIUS,
  TIME_PERIODS,
}; 