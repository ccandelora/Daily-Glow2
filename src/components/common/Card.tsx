import React from 'react';
import { View, StyleSheet, ViewStyle } from 'react-native';
import theme from '../../constants/theme';

interface CardProps {
  children: React.ReactNode;
  style?: ViewStyle;
  variant?: 'default' | 'elevated';
}

export const Card: React.FC<CardProps> = ({
  children,
  style,
  variant = 'default',
}) => {
  return (
    <View
      style={[
        styles.card,
        variant === 'elevated' && styles.elevated,
        style,
      ]}
    >
      {children}
    </View>
  );
};

const styles = StyleSheet.create({
  card: {
    backgroundColor: theme.COLORS.ui.card,
    borderRadius: theme.BORDER_RADIUS.lg,
    padding: theme.SPACING.md,
    marginVertical: theme.SPACING.sm,
  },
  elevated: {
    shadowColor: theme.COLORS.ui.text,
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    elevation: 5,
  },
}); 