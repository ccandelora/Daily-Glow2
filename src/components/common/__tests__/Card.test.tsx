import React from 'react';
import { render } from '@testing-library/react-native';
import { Card } from '../Card';
import { Text, StyleSheet } from 'react-native';
import theme from '@/constants/theme';

describe('Card Component', () => {
  test('renders correctly with default props', () => {
    const { toJSON } = render(
      <Card>
        <Text>Test Content</Text>
      </Card>
    );
    
    expect(toJSON()).toMatchSnapshot();
  });
  
  test('renders with correct default styles', () => {
    const { getByTestId } = render(
      <Card testID="card-container">
        <Text>Test Content</Text>
      </Card>
    );
    
    const cardContainer = getByTestId('card-container');
    
    // Check specific style properties instead of the entire style object
    expect(cardContainer.props.style[0]).toMatchObject({
      backgroundColor: theme.COLORS.ui.card,
      borderRadius: theme.BORDER_RADIUS.lg,
      padding: theme.SPACING.lg,
    });
  });
  
  test('applies the elevated style when variant is elevated', () => {
    const { getByTestId } = render(
      <Card variant="elevated" testID="card-container">
        <Text>Elevated Card</Text>
      </Card>
    );
    
    const cardContainer = getByTestId('card-container');
    
    // Testing specific style properties from the elevated style
    expect(cardContainer.props.style[0]).toMatchObject({
      backgroundColor: theme.COLORS.ui.card,
    });
    expect(cardContainer.props.style[1]).toMatchObject({
      backgroundColor: theme.COLORS.ui.background,
      borderColor: theme.COLORS.ui.accent,
    });
  });
  
  test('applies the glow style when variant is glow', () => {
    const { getByTestId } = render(
      <Card variant="glow" testID="card-container">
        <Text>Glowing Card</Text>
      </Card>
    );
    
    const cardContainer = getByTestId('card-container');
    
    // Testing specific style properties from the glow style
    expect(cardContainer.props.style[0]).toMatchObject({
      backgroundColor: theme.COLORS.ui.card,
    });
    expect(cardContainer.props.style[2]).toMatchObject({
      backgroundColor: 'rgba(38, 20, 60, 0.95)',
      borderColor: theme.COLORS.ui.accent,
    });
  });
  
  test('applies custom styles when provided', () => {
    const customStyle = {
      backgroundColor: 'purple',
      margin: 20,
    };
    
    const { getByTestId } = render(
      <Card style={customStyle} testID="card-container">
        <Text>Custom Styled Card</Text>
      </Card>
    );
    
    const cardContainer = getByTestId('card-container');
    
    // Custom styles should be applied as the fourth item in the style array
    expect(cardContainer.props.style[3]).toMatchObject(customStyle);
  });
  
  test('passes additional props to View component', () => {
    const accessibilityLabel = 'Card component with accessibility label';
    const { getByTestId } = render(
      <Card testID="card-container" accessibilityLabel={accessibilityLabel}>
        <Text>Accessible Card</Text>
      </Card>
    );
    
    const cardContainer = getByTestId('card-container');
    expect(cardContainer.props.accessibilityLabel).toBe(accessibilityLabel);
  });
}); 