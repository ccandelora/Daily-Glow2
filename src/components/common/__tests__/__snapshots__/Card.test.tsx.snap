// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component renders correctly with default props 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "rgba(38, 20, 60, 0.85)",
        "borderColor": "rgba(82, 67, 194, 0.3)",
        "borderRadius": 16,
        "borderWidth": 1,
        "elevation": 3,
        "padding": 24,
        "shadowColor": "#4169E1",
        "shadowOffset": {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.15,
        "shadowRadius": 8,
      },
      false,
      false,
      undefined,
    ]
  }
>
  <Text>
    Test Content
  </Text>
</View>
`;
